diff --git a/vendor/amasty/module-request-quote/Block/Adminhtml/Items/AbstractItems.php b/vendor/amasty/module-request-quote/Block/Adminhtml/Items/AbstractItems.php
index be7bfd1ce721f81b9d6cd849693978e165d9a4dc..c9c78af4bc136407c9e6da7b3c329f38c2db135f 100644
--- a/vendor/amasty/module-request-quote/Block/Adminhtml/Items/AbstractItems.php
+++ b/vendor/amasty/module-request-quote/Block/Adminhtml/Items/AbstractItems.php
@@ -196,14 +196,10 @@ class AbstractItems extends \Magento\Backend\Block\Template
         );
     }

-    /**
-     * @param bool $strong
-     * @param string $separator
-     * @return string
-     */
-    public function displayProductPrice($strong = false, $separator = '<br />')
+    public function displayProductPrice(int $qty = 1, bool $strong = false, string $separator = '<br />'): string
     {
-        $productPrice = $this->getPriceDataObject()->getProduct()->getFinalPrice();
+        $productPrice = $this->getPriceDataObject()->getProduct()->getFinalPrice($qty);
+
         return $this->displayPrices(
             $productPrice,
             $this->getBaseCurrency()->convert($productPrice, $this->getCurrency()),
diff --git a/vendor/amasty/module-request-quote/Block/Adminhtml/Quote/View/Items/Renderer/DefaultRenderer.php b/vendor/amasty/module-request-quote/Block/Adminhtml/Quote/View/Items/Renderer/DefaultRenderer.php
index f263bf7e3807e99c6f23d474b7295c8baac160b4..d303acd9adde9d0f4574badc030ab2ed55647971 100644
--- a/vendor/amasty/module-request-quote/Block/Adminhtml/Quote/View/Items/Renderer/DefaultRenderer.php
+++ b/vendor/amasty/module-request-quote/Block/Adminhtml/Quote/View/Items/Renderer/DefaultRenderer.php
@@ -146,7 +146,9 @@ class DefaultRenderer extends \Amasty\RequestQuote\Block\Adminhtml\Items\Rendere
                 $html = $this->displayProductPrice();
                 break;
             case 'price':
-                $html = $item->getCustomPrice() ? $this->displayCustomPrice() : $this->displayProductPrice();
+                $html = $item->getCustomPrice()
+                    ? $this->displayCustomPrice()
+                    : $this->displayProductPrice($item->getQty());
                 break;
             case 'qty':
                 $html = $item->getQty() * 1;
diff --git a/vendor/amasty/module-request-quote/Model/Quote/Backend/Edit/ProductManagement.php b/vendor/amasty/module-request-quote/Model/Quote/Backend/Edit/ProductManagement.php
index bd0c401c57f218b7918d6f12fa74c1b04a669f60..b692b0d517985621b24a6ecaced6896b52432c31 100644
--- a/vendor/amasty/module-request-quote/Model/Quote/Backend/Edit/ProductManagement.php
+++ b/vendor/amasty/module-request-quote/Model/Quote/Backend/Edit/ProductManagement.php
@@ -187,10 +187,14 @@ class ProductManagement
                 }
                 $info['qty'] = (double)$info['qty'];
             }
-            $info['price'] = (double)$info['price'];
-            if ($this->isResetPriceModificators()) {
-                $info['price'] = $this->getRequestedPrice($item);
+
+            if (isset($info['price'])) {
+                $info['price'] = (double)$info['price'];
+                if ($this->isResetPriceModificators()) {
+                    $info['price'] = $this->getRequestedPrice($item);
+                }
             }
+
             $info['modificators'] = [
                 QuoteInterface::SURCHARGE => $this->getSurcharge(),
                 QuoteInterface::DISCOUNT => $this->getDiscount()
diff --git a/vendor/amasty/module-request-quote/Model/Quote/Item/Updater.php b/vendor/amasty/module-request-quote/Model/Quote/Item/Updater.php
index ec8a2e0e560073d3dee538e83dff2a711e76c708..72bf6147af02e2d494c5cd8bbbcff09d8d1be5df 100644
--- a/vendor/amasty/module-request-quote/Model/Quote/Item/Updater.php
+++ b/vendor/amasty/module-request-quote/Model/Quote/Item/Updater.php
@@ -100,8 +100,6 @@ class Updater
             /** @var \Magento\Framework\DataObject $infoBuyRequest */
             $infoBuyRequest = $item->getBuyRequest();
             if ($infoBuyRequest) {
-                $infoBuyRequest->setPrice($itemPrice);
-
                 $infoBuyRequest->setValue($this->serializer->serialize($infoBuyRequest->getData()));
                 $infoBuyRequest->setCode('info_buyRequest');
                 $infoBuyRequest->setProduct($item->getProduct());
diff --git a/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/grid.phtml b/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/grid.phtml
index d6fe8c7298ea4d6270f65649dbaac4a9017f5dd9..09bfdd5b7adec645f0da2f44c3ef6a781e549abd 100644
--- a/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/grid.phtml
+++ b/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/grid.phtml
@@ -100,7 +100,7 @@ $priceModifiersBlock = $block->getChildBlock('price_modifiers');
                                name="item[<?= $escaper->escapeHtml($_item->getId()) ?>][price]"
                                value="<?= $escaper->escapeHtml(sprintf("%.2f", $block->getItemEditablePrice($_item))) ?>"
                             <?php if (!$_isCustomPrice): ?>
-                                style="display:none"
+                                style="display:none" disabled
                             <?php endif; ?>
                                class="input-text item-price admin__control-text"/>
                     </td>
diff --git a/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/price/unit.phtml b/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/price/unit.phtml
index 5f727c162f82100cc06db2cd5c3502a3453800ae..7988d3b859f94ecfbd923369f01b648c79b53855 100644
--- a/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/price/unit.phtml
+++ b/vendor/amasty/module-request-quote/view/adminhtml/templates/quote/create/items/price/unit.phtml
@@ -3,5 +3,5 @@
 $_item = $block->getItem();
 ?>
 <div class="price-excl-tax">
-<?= /* @noEscape */ $block->formatPrice($_item->getProduct()->getFinalPrice(), true) ?>
+<?= /* @noEscape */ $block->formatPrice($_item->getProduct()->getFinalPrice($_item->getQty()), true) ?>
 </div>
diff --git a/vendor/amasty/module-request-quote/view/adminhtml/web/quote/edit/scripts.js b/vendor/amasty/module-request-quote/view/adminhtml/web/quote/edit/scripts.js
index 3b80a4476e8eb7eef423bdbaa675e5dd025a9553..f119a7325ef778a0bf54703d49770f98adce5eb1 100644
--- a/vendor/amasty/module-request-quote/view/adminhtml/web/quote/edit/scripts.js
+++ b/vendor/amasty/module-request-quote/view/adminhtml/web/quote/edit/scripts.js
@@ -1108,9 +1108,11 @@ define([

         toggleCustomPrice: function(checkbox, elemId) {
             if (checkbox.checked) {
+                $(elemId).disabled = false;
                 $(elemId).show();
             }
             else {
+                $(elemId).disabled = true;
                 $(elemId).hide();
             }
         },
