diff --git a/vendor/magento/module-sales-rule/Model/Quote/Discount.php b/vendor/magento/module-sales-rule/Model/Quote/Discount.php
index a32fe2499..008d12501 100644
--- a/vendor/magento/module-sales-rule/Model/Quote/Discount.php
+++ b/vendor/magento/module-sales-rule/Model/Quote/Discount.php
@@ -161,7 +161,7 @@ class Discount extends AbstractTotal
             $eventArgs['item'] = $item;
             $this->eventManager->dispatch('sales_quote_address_discount_item', $eventArgs);
 
-            if ($item->getHasChildren() && $item->isChildrenCalculated()) {
+            if ($item->getHasChildren() && $item->isChildrenCalculated() && $item->getProduct()->getTypeId() != 'bundle') {
                 $this->calculator->process($item);
                 foreach ($item->getChildren() as $child) {
                     $eventArgs['item'] = $child;
