diff --git a/vendor/avalara/excise/Observer/Sales/OrderInvoicePayBefore.php b/vendor/avalara/excise/Observer/Sales/OrderInvoicePayBefore.php
Index: ../vendor/avalara/excise/Observer/Sales/OrderInvoicePayBefore.php
--- a/vendor/avalara/excise/Observer/Sales/OrderInvoicePayBefore.php	2022-05-06 04:19:39.180108097 -0400
+++ b/vendor/avalara/excise/Observer/Sales/OrderInvoicePayBefore.php	2022-05-06 04:21:58.616044952 -0400
@@ -38,12 +38,12 @@
             $invoiceItemExciseTax = $invoiceItemSalesTax = 0;
             if ($orderItem->getExciseTax() > 0) {
                 $orderItemExciseTax = $orderItem->getExciseTax() / $orderItem->getQtyOrdered();
-                $invoiceItemExciseTax = number_format($orderItemExciseTax * $item->getQty(), 2);
+                $invoiceItemExciseTax = number_format($orderItemExciseTax * $item->getQty(), 2, '.', '');
             }
 
             if ($orderItem->getSalesTax() > 0) {
                 $orderItemSalesTax = $orderItem->getSalesTax() / $orderItem->getQtyOrdered();
-                $invoiceItemSalesTax = number_format($orderItemSalesTax * $item->getQty(), 2);
+                $invoiceItemSalesTax = number_format($orderItemSalesTax * $item->getQty(), 2, '.', '');
             }
 
             $invoiceExciseTax += $invoiceItemExciseTax;
