diff --git a/vendor/avalara/excise/Model/ProcessTaxQuote.php b/vendor/avalara/excise/Model/ProcessTaxQuote.php
Index: ../vendor/avalara/excise/Model/ProcessTaxQuote.php
--- a/vendor/avalara/excise/Model/ProcessTaxQuote.php	2022-05-31 19:01:31.000000000 +0530
+++ b/vendor/avalara/excise/Model/ProcessTaxQuote.php	2022-10-18 03:47:34.852236132 +0530
@@ -973,7 +973,7 @@
                 /** @var \Magento\Catalog\Model\Product $product */
                 $product = $item->getProduct();
 
-                if ($item->getProductType() == "bundle") {
+                if ($item->getProductType() == "bundle" || $item->getProductType() == "configurable") {
                     continue;
                 }
 
@@ -981,9 +981,11 @@
                 $currencyCode = $this->storeManager->getStore()->getCurrentCurrency()->getCode();
 
                 if ($item->getParentItem()) {
-                    if ($item->getParentItem()->getProductType() == "configurable") {
-                        continue;
-                    }
+                    /*
+                        if ($item->getParentItem()->getProductType() == "configurable") {
+                            continue;
+                        }
+                    */
                     $itemQty = $item->getParentItem()->getQty();
                 } else {
                     $itemQty = $item->getQty();
