diff --git a/vendor/avalara/excise/Model/ProcessTaxQuote.php b/vendor/avalara/excise/Model/ProcessTaxQuote.php
index a73fdffa..007ee862 100644
--- a/vendor/avalara/excise/Model/ProcessTaxQuote.php
+++ b/vendor/avalara/excise/Model/ProcessTaxQuote.php
@@ -1003,6 +1003,9 @@ class ProcessTaxQuote
 
                 $itemSku = $item->getSku();
                 $itemUnitPrice = $item->getPrice();
+                if($itemUnitPrice <= 0) {
+			continue;
+		}
                 //$itemUnitPrice = $item->getRowTotal();
                 $itemAlternateUnitPrice = $item->getProduct()->getExcisePurchaseUnitPrice();
                 $itemUnitOfMeasure = ($item->getProduct()->getAttributeText('excise_unit_of_measure')) ? $item->getProduct()->getAttributeText('excise_unit_of_measure') : "PAK";
