diff --git a/vendor/amzn/amazon-pay-and-login-with-amazon-core-module/Block/Adminhtml/Form/Field/IpnUrl.php b/vendor/amzn/amazon-pay-and-login-with-amazon-core-module/Block/Adminhtml/Form/Field/IpnUrl.php
index b6895c56..e71fc14e 100755
--- a/vendor/amzn/amazon-pay-and-login-with-amazon-core-module/Block/Adminhtml/Form/Field/IpnUrl.php
+++ b/vendor/amzn/amazon-pay-and-login-with-amazon-core-module/Block/Adminhtml/Form/Field/IpnUrl.php
@@ -41,12 +41,13 @@ class IpnUrl extends BaseField
         $store = $this->_storeManager->getDefaultStoreView();
         $valueReturn = '';
 
-        $baseUrl = $store->getBaseUrl(UrlInterface::URL_TYPE_WEB, true);
-        if ($baseUrl) {
-            $value       = $baseUrl . 'amazonpayments/payment/ipn/';
-            $valueReturn = "<div>".$this->escapeHtml($value)."</div>";
+        if ($store) {
+            $baseUrl = $store->getBaseUrl(UrlInterface::URL_TYPE_WEB, true);
+            if ($baseUrl) {
+                $value       = $baseUrl . 'amazonpayments/payment/ipn/';
+                $valueReturn = "<div>".$this->escapeHtml($value)."</div>";
+            }
         }
-
         return '<td class="value">' . $valueReturn . '</td>';
     }
 
