diff --git a/app/code/Magedelight/Subscribenow/Setup/UpgradeData.php b/app/code/Magedelight/Subscribenow/Setup/UpgradeData.php
index eb63f9ca..f3e5a03d 100644
--- a/app/code/Magedelight/Subscribenow/Setup/UpgradeData.php
+++ b/app/code/Magedelight/Subscribenow/Setup/UpgradeData.php
@@ -74,10 +74,6 @@ class UpgradeData implements UpgradeDataInterface
             $this->columnDataUpgradeFactory->create()->upgradeData($setup, $sales);
         }
 
-        if (version_compare($context->getVersion(), '200.4.0', '<')) {
-            $this->mysqlCompatibilityUpgradeFactory->create()->upgradeData($setup, $sales);
-        }
-
         if (version_compare($context->getVersion(), '200.5.0', '<')) {
             $this->upgrade200x5x0Factory->create()->upgradeData($setup, $context, $sales);
         }
