diff --git a/app/code/Tabby/Checkout/Observer/ConfigObserver.php b/app/code/Tabby/Checkout/Observer/ConfigObserver.php
index 6fe00af31..80635701a 100644
--- a/app/code/Tabby/Checkout/Observer/ConfigObserver.php
+++ b/app/code/Tabby/Checkout/Observer/ConfigObserver.php
@@ -88,18 +88,18 @@ class ConfigObserver implements ObserverInterface
                 continue;
             }
             if ($this->isMethodActive($store->getId())) {
-                if (!array_key_exists($store->getGroupId(), $register_hooks)) {
-                    $register_hooks[$store->getGroupId()] = [];
+                if (!array_key_exists($store->getId(), $register_hooks)) {
+                    $register_hooks[$store->getId()] = [];
                 }
-                $register_hooks[$store->getGroupId()] = array_unique(array_merge(
-                    $register_hooks[$store->getGroupId()],
+                $register_hooks[$store->getId()] = array_unique(array_merge(
+                    $register_hooks[$store->getId()],
                     $store->getAvailableCurrencyCodes()
                 ));
             }
         }
-        foreach ($register_hooks as $groupId => $currencies) {
-            $group = $this->_storeManager->getGroup($groupId);
-            $webhookUrl = $this->_urlHelper->getUrl('tabby/result/webhook', ['_scope' => $group->getDefaultStoreId()]);
+        foreach ($register_hooks as $storeId => $currencies) {
+            $store = $this->_storeManager->getStore($storeId);
+            $webhookUrl = $this->_urlHelper->getUrl('tabby/result/webhook', ['_scope' => $store->getId()]);
 
             if ($this->getWebsiteConfigValue('tabby/tabby_api/local_currency', $website->getCode())) {
                 $currencies = array_unique($currencies);
@@ -109,13 +109,13 @@ class ConfigObserver implements ObserverInterface
                         continue;
                     }
                     $this->_api->registerWebhook(
-                        $group->getDefaultStoreId(),
-                        $group->getCode() . '_' . $currencyCode,
+                        $store->getId(),
+                        $store->getCode() . '_' . $currencyCode,
                         $webhookUrl
                     );
                 }
             } else {
-                $this->_api->registerWebhook($group->getDefaultStoreId(), $group->getCode(), $webhookUrl);
+                $this->_api->registerWebhook($store->getId(), $store->getCode(), $webhookUrl);
             }
         }
     }
